readcomb/phase_change_filter.py

```diff
@@ -42,151 +34,270 @@ def check_snps(f_name, chromosome, left_bound, right_bound):
-                # shift the start over by the amount of insertion to compensate for it
+                # shift the start to the right by the amount of insertion to compensate for it
                 start += cigar_tuples[current_tuple][1]
             
+            elif cigar_tuples[current_tuple][0] == 2:
+                # shift the start to the left by the amount of deletion to compensate for it
+                start -= cigar_tuples[current_tuple][1]
+
             current_base += cigar_tuples[current_tuple][1]
             current_tuple += 1
-    
+
+
         # indexing for VCF seems to be a bit weird and will sometimes be -1
+        # this has been fixed but left check in anyways
         if start < 0:
             raise Exception('VCF indexing is off. Check SNP at {}'.format(snp))
-        
+
         strand1 = snp.gt_bases[0][0]
         strand2 = snp.gt_bases[1][0]
-        
+
         if start >= len(segment):
             break
-        
+
         if segment[start] == strand1:
-            snp_lst[start] = '1'
+            snp_lst.append('1')
+
         elif segment[start] == strand2:
-            snp_lst[start] = '2'
+            snp_lst.append('2')
+
         else:
-            snp_lst[start] = 'N'
+            snp_lst.append('N')
+
+    return snp_lst
+
+def matepairs_recomb():
+    '''
+    Given a bam file object, return a human-readable file with aligned reference, quality string, 
+        bam sequence, and SNPs    
+
+    If mode='no_match', create a file with only no_match sequences.
+    If mode='all', create a file with all sequences
+    If mode='phase_change', create a file with only sequences with phase changes
+    
+    Makes assumptions on the following file locations:
+        VCF: vcf_files_path + 'parental_filtered.vcf.gz'
+        Reference: reference_files_path + 'chlamy.5.3.w_organelles_mtMinus.fasta'
+    '''
+
+    bam, vcf, mode, output_filename = args()
+
+    bam_file_obj = pysam.AlignmentFile(bam, 'r')
+    
+    f_obj = pysam.AlignmentFile(output_filename + '.sam', 'wh', template=bam_file_obj)
+    
+    #counters
+    no_match_counter = 0
+    phase_change_counter = 0
+    phase_change_mate_pair_counter = 0
+    seq_with_snps_counter = 0
+    all_seq_counter = 0
+    
+    # get mate pairs
+    pairs = cache_pairs(bam_file_obj)
+    
+    for query_name in pairs:
+        
+        all_seq_counter += 1
+        
+        # initialize first record
+        record = pairs[query_name][0]
+        
+        # analyze cigar string
+        segment = cigar(record)
+                    
+        snps = check_snps(vcf, record.reference_name, 
+                        record.reference_start,
+                        record.reference_start + record.query_alignment_length)
+        
+        snp_lst = phase_detection(snps, segment, record)
+        
+        # initialize second record if there is a matepair
+        if pairs[query_name][1]:
+            
+            seq_with_snps_counter += 1
```

Comment:
Is this counter specifically to count the number of reads with pairs? Bc if so it should have a different name

Reply:
Oh yeah, that's a relic left over from before I overhauled matepair_recomb, will update this